azurerm/internal/services/containers/kubernetes_cluster_pod_identity_resource_test.go

```diff
@@ -0,0 +1,239 @@
+		!*resp.ManagedClusterProperties.PodIdentityProfile.Enabled {
+		return utils.Bool(false), nil
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r KubernetesClusterPodIdentityResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  depends_on = [
+    azurerm_role_assignment.test
+  ]
+}
+`, r.template(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "import" {
+  cluster_id = azurerm_kubernetes_cluster_pod_identity.test.cluster_id
+
+  dynamic "pod_identity" {
+    for_each = azurerm_kubernetes_cluster_pod_identity.test.pod_identity
+    content {
+      name        = pod_identity.value.name
+      namespace   = pod_identity.value.namespace
+      identity_id = pod_identity.value.identity_id
+    }
+  }
+}
+`, r.basic(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_kubernetes_cluster_pod_identity" "test" {
+  cluster_id = azurerm_kubernetes_cluster.test.id
+
+  pod_identity {
+    name        = "name1"
+    namespace   = "ns1"
+    identity_id = azurerm_user_assigned_identity.test.id
+  }
+
+  exception {
+    name      = "exception1"
+    namespace = "exception-ns1"
+    pod_labels = {
+      "env" : "test"
+    }
+  }
+
+  depends_on = [
+    azurerm_role_assignment.test
+  ]
+}
+`, r.template(data))
+}
+
+func (r KubernetesClusterPodIdentityResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-aks-%d"
+  location = "%s"
+}
+
+resource "azurerm_kubernetes_cluster" "test" {
+  name                = "acctestaks%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  dns_prefix          = "acctestaks%d"
+
+  linux_profile {
+    admin_username = "acctestuser%d"
+
+    ssh_key {
+      key_data = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCqaZoyiz1qbdOQ8xEf6uEu1cCwYowo5FHtsBhqLoDnnp7KUTEBN+L2NxRIfQ781rxV6Iq5jSav6b2Q8z5KiseOlvKA/RF2wqU0UPYqQviQhLmW6THTpmrv/YkUCuzxDpsH7DUDhZcwySLKVVe0Qm3+5N2Ta6UYH3lsDf9R9wTP2K/+vAnflKebuypNlmocIvakFWoZda18FOmsOoIVXQ8HWFNCuw9ZCunMSN62QGamCe3dL5cXlkgHYv7ekJE15IA9aOJcM7e90oeTqo+7HTcWfdu0qQqPWY5ujyMw/llas8tsXY85LFqRnr3gJ02bAscjc477+X+j/gkpFoN1QEmt terraform@demo.tld"
+    }
+  }
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_DS2_v2"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  network_profile {
```

Comment:
Is this required for the test?

Reply:
yes

Reply:
Persumably this is missing in the document?